gcloud/resource_manager/project.py

```diff
@@ -0,0 +1,323 @@
+            >>> client = resource_manager.Client()
+            >>> project = client.project('purple-spaceship-456')
+            >>> project.exists()
+            False
+
+        You can also use the
+        :func:`gcloud.resource_manager.client.Client.get_project`
+        method to check whether a project exists, as it will return ``None``
+        if the project doesn't exist::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> print client.get_project('purple-spaceship-456')
+            None
+
+        See
+        https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/projects/get
+
+        :type client: :class:`gcloud.resource_manager.client.Client` or None
+        :param client: the client to use.  If not passed, falls back to
+                       the ``client`` attribute.
+        """
+        client = self._require_client(client=client)
+
+        try:
+            client.connection.api_request(method='GET', path=self.path)
+        except NotFound:
+            return False
+        else:
+            return True
+
+    def delete(self, client=None, reload_data=True):
+        """API call:  delete the project via a ``DELETE`` request.
+
+        See:
+        https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/delete
+
+        This actually changes the status (``lifecycleState``) from ``ACTIVE``
+        to ``DELETE_REQUESTED``.
+        Later (it's not specified when), the project will move into the
+        ``DELETE_IN_PROGRESS`` state, which means the deleting has actually
+        begun.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> project = client.get_project('purple-spaceship-123')
+            >>> project.delete()
+
+        :type client: :class:`gcloud.resource_manager.client.Client` or None
+        :param client: the client to use.  If not passed,
+                       falls back to the ``client`` attribute.
+
+        :type reload_data: bool
+        :param reload_data: Whether to reload the project with the latest
+                            state. Default: ``True``.
+        """
+        client = self._require_client(client)
+        client.connection.api_request(method='DELETE', path=self.path)
+
+        # If the reload flag is True, reload the project.
+        if reload_data:
+            self.reload()
+
+    def undelete(self, client=None, reload_data=True):
+        """API call:  undelete the project via a ``POST`` request.
+
+        See
+        https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/undelete
+
+        This actually changes the project status (``lifecycleState``) from
+        ``DELETE_REQUESTED`` to ``ACTIVE``.
+        If the project has already reached a status of ``DELETE_IN_PROGRESS`,
+        this request will fail and the project cannot be restored.
```

Comment:
See above in `delete()`. ("Later (it's not specified when) ..."

This one changes from `DELETE_REQUESTED` to `ACTIVE` immediately.